test/report.test.ts

```diff
@@ -0,0 +1,59 @@
+// (C) 2007-2018 GoodData Corporation
+import 'isomorphic-fetch';
+import * as fetchMock from 'fetch-mock';
+import { ReportModule } from '../src/report';
+import { XhrModule } from '../src/xhr';
+import { IExportConfig } from '../src/interfaces';
+import { mockPollingRequest } from './helpers/polling';
+
+const mockedReportModule = () => new ReportModule(new XhrModule(fetch, {}));
+
+describe('report', () => {
+    const projectId = '1';
+    const projectUri = `/gdc/internal/projects/${projectId}/exportResult`;
+
+    describe('export', () => {
+        afterEach(() => {
+            fetchMock.restore();
+        });
+
+        describe('exportResult', () => {
+            it('should return created file', () => {
+                const createdReport = '/gdc/exporter/result/12345';
+                fetchMock.mock(
+                    projectUri,
+                    {
+                        status: 200,
+                        body: { uri: createdReport }
+                    }
+                );
+
+                const mockTask = (status: any) => ({ status });
+                const finishedTask = mockTask('OK');
+                const runningTask = mockTask('RUNNING');
```

Comment:
looks like mockTask responses do not match `isDataExported` test in handlePooling